### ISSUE TITLE
[attr.width.px] can't work in angular9

### ISSUE THREAD

**Issue by DreamLi1314** [angular/angular#36256]
**Env**
* os: win10
* browser: Chrome 80.0.3987.149（x64)
* `package.json`
```json
    "@angular/animations": "^9.0.7",
    "@angular/cdk": "^9.1.3",
    "@angular/common": "^9.0.7",
    "@angular/compiler": "^9.0.7",
    "@angular/core": "^9.0.7",
    "@angular/forms": "^9.0.7",
    "@angular/localize": "^9.0.7",
    "@angular/material": "^9.1.3",
    "@angular/material-moment-adapter": "^9.1.3",
    "@angular/platform-browser": "^9.0.7",
    "@angular/platform-browser-dynamic": "^9.0.7",
    "@angular/router": "^9.0.7"
```

**Desc**
``` typescript
<td *ngIf="hasContent(r, c)"
    [attr.width.px]="model?.cells[r][c]?.width"
    [attr.height.px]="model?.cells[r][c]?.height + (isEditableMode() && last ? 1 : 0)"
```

![Browser Display](https://user-images.githubusercontent.com/23147720/77643877-128bfe80-6f9b-11ea-86df-f6c15b239f36.png)

Rendering failed because the property was compiled to `width.px`

**Comment by DreamLi1314** [comment]
`[style.width.px]="model?.cells[r][c]?.width"` and `[attr.width]="model?.cells[r][c]?.width"` is work for me.
But I think this should be a mistake or why doesn't this work in Angular9?

**Comment by pkozlowski-opensource** [comment]
tl;dr; this is _not_ the valid syntax and it just "happened to work" in previous versions of Angular. As @DreamLi1314 suggested you need to use one of:
* `[style.width.px]="expression"` if you want to bind to the [CSS style property](https://developer.mozilla.org/en-US/docs/Web/CSS/width) (probably this is what you want);
* `[att.width]="expression + 'px'"` if you want to bind to the [width attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes)

What was happening with the VE attribute parser that any suffix would be dropped without any warning ( https://stackblitz.com/edit/angular-9qzvjk) which isn't greatest user experience... I'm going to keep this issue open so we can decide what to do here (probably error when detecting such usage pattern).

**Comment by DreamLi1314** [comment]
ok, thanks.

**Comment by pkozlowski-opensource** [comment]
Thinking about it a bit more, I would say that, as the very minimum, we should have compile-time error when encountering `[attr.width.px]` syntax (that is, we should verify that the attribute binding syntax comes only in the `[attr.attrname]` flavour and is _not_ accepting suffix.

**Comment by vikassrivastava18** [comment]
### Not working in Angular 11
**[style.width.px]="expression"**

<div class="crop"
     [style.width.px]="starWidth">
  <div style="width: 75px">
    <span class="fa fa-star"></span>
    <span class="fa fa-star"></span>
    <span class="fa fa-star"></span>
    <span class="fa fa-star"></span>
    <span class="fa fa-star"></span>
  </div>
</div>
